### DISCUSSION TITLE
metadata of basemap

### DISCUSSION THREAD

**Discussion by MananSarupria599** [opengeos/leafmap#701] [Q&A] [answered]
I am utilizing the SAM package to segment farmlands, with the 'Satellite' basemap serving as my primary imagery source. I am interested in determining the time/year when the 'satellite' basemap used for segmentation was captured.

**Comment by giswqs** [comment, marked as answer]
I don't think it is possible to get the medata of imagery diretcly through coding. One workdaround is to use Google Earth Pro os to identify imagery acquation date.

![image](https://github.com/opengeos/leafmap/assets/5016453/e435b71c-095e-4805-9ee9-f6f9813c77e4)

**Comment by SFrav** [comment]
In case others would like to find the imagery source and acquisition date for many scenes programmatically, I imagine that it could be accessed via the earth.google.com website. This would rely on selenium, among packages. Looking at the site I don't see any easy way to do this as all buttons and text are embedded in what is presumably a JS application. I imagine that native selenium and JS options will not work, unfortunately. Instead, one might have to use pyautogui to automate clicks. See below for example pseudo code.

```
from selenium import webdriver
import pyautogui
import pyperclip
import time

lat = "37.422"
lon = "-122.084"

driver = webdriver.Chrome()
driver.get('https://earth.google.com/web/@'+lat+','+lon+',1.1a,30000d')

#Click attribution link at the bottom of the screen
#Use pyautogui.position() to find the location of the last click
pyautogui.click(x, y) #insert x and y values from position identified earlier 

#... then run other pyautogui functions to extract the highlightable text.
# As a last resort this kind of combination will most likely work pyautogui.click(x1, y1 clicks=3)  ... pyautogui.rightClick(x1, y1) ... pyautogui.press('down') ... pyautogui.press('enter') # to highlight the text and then copy using the right-click menu. 

data = pyperclip.paste()
print("Data:", data)
```

Be sure to view the terms of service beforehand.
